cmd/loadTests.go

```diff
@@ -53,6 +55,102 @@ var (
 	threadsWG                          sync.WaitGroup
 )
 
+type LogData struct {
+	Timestamp                         string      `json:"timestamp"`
+	MachineName                       string      `json:"machineName"`
+	BinaryDetails                     string      `json:"binaryDetails"`
+	NumberOfThreads                   int         `json:"Number of threads"`
+	NumberOfUsersPerThread            int         `json:"Number of users per thread"`
+	BatchSize                         int         `json:"Batch size per thread"`
+	NumberOfUsers                     int         `json:"Total number of users"`
+	LoadTestCompletionStatus          string      `json:"loadTestCompletionStatus"`
+	AverageTimeToSpinUpUsers          FloatFormat `json:"Average Time taken to spin up users (sec)"`
+	AverageTimeToCreateResources      FloatFormat `json:"Average Time taken to Create Resources (sec)"`
+	AverageTimeToRunPipelines         FloatFormat `json:"Average Time taken to Run Pipelines (sec)"`
+	UserCreationFailureCount          int64       `json:"Number of times user creation failed"`
+	UserCreationFailurePercentage     FloatFormat `json:"User creation failed percentage"`
+	ResourceCreationFailureCount      int64       `json:"Number of times resource creation failed"`
+	ResourceCreationFailurePercentage FloatFormat `json:"Resource creation failed percentage"`
+	PipelineRunFailureCount           int64       `json:"Number of times pipeline run failed"`
+	PipelineRunFailurePercentage      FloatFormat `json:"Pipeline run failed percentage"`
+}
+
+// to marshall json float as .2f% we need the MarshalJSON function
+
+type FloatFormat float64
+
+func (f FloatFormat) MarshalJSON() ([]byte, error) {
+	return json.Marshal(fmt.Sprintf("%.2f", f))
+}
+
+func createLogDataJSON(
+	outputFile string,
+	timestamp string,
+	numberOfUsers int,
+	numberOfThreads int,
+	numberOfUsersPerThread int,
+	batchSize int,
+	loadTestCompletionStatus string,
+	averageTimeToSpinUpUsers float64,
+	averageTimeToCreateResources float64,
+	averageTimeToRunPipelines float64,
+	userCreationFailureCount int64,
+	userCreationFailurePercentage float64,
+	resourceCreationFailureCount int64,
+	resourceCreationFailurePercentage float64,
+	pipelineRunFailureCount int64,
+	pipelineRunFailurePercentage float64,
```

Comment:
@naftalysh Instead of passing all the json data field values as function arguments, could you modify the function to pass the `LogData` structure as a single argument? That will be more maintainable in the future when we add fields to the data. Thanks

Reply:
That way you can keep an instance of the logData by the test, modify it as needed even during the load test execution and then just pass it to the function that renders the JSON file when needed.